### DISCUSSION TITLE
자바스크립트에서 변수의 생명 주기(lifecycle)에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#24] [변수]
- 자바스크립트에서 변수의 생명 주기(lifecycle)에 대해 설명해 주세요.

**Comment by publdaze** [comment]
> - 자바스크립트에서 변수의 생명 주기(lifecycle)에 대해 설명해 주세요.
- 변수가 생성되고 소멸되는 시간 동안의 범위
- 선언 -> 초기화 -> 소멸 단계
- 지역 변수의 생명주기와 전역 변수의 생명주기가 다름

**Comment by pipisebastian** [comment]
> 자바스크립트에서 변수의 생명 주기(lifecycle)에 대해 설명해 주세요.

변수가 메모리에 할당되고, 해제될때까지의 시간입니다.
